cdap-integration-test/src/main/java/co/cask/cdap/test/IntegrationTestBase.java

```diff
@@ -312,13 +339,41 @@ protected ApplicationManager deployApplication(Id.Namespace namespace,
   }
 
   protected ApplicationManager deployApplication(Class<? extends Application> applicationClz) throws IOException {
-    return deployApplication(Id.Namespace.DEFAULT, applicationClz);
+    return deployApplication(getConfiguredNamespace().toId(), applicationClz);
   }
 
   protected ApplicationManager getApplicationManager(ApplicationId applicationId) throws Exception {
     return getTestManager().getApplicationManager(applicationId);
   }
 
+  private void doClear(NamespaceId namespace) throws Exception {
+    // stop all programs in the namespace
+    getProgramClient().stopAll(namespace.toId());
+    // delete all apps in the namespace
+    for (ApplicationRecord app : getApplicationClient().list(namespace.toId())) {
+      getApplicationClient().delete(namespace.app(app.getName()).toId());
+    }
+    // delete all streams
+    for (StreamDetail streamDetail : getStreamClient().list(namespace.toId())) {
+      getStreamClient().delete(namespace.stream(streamDetail.getName()).toId());
+    }
+    // delete all dataset instances
+    for (DatasetSpecificationSummary datasetSpecSummary : getDatasetClient().list(namespace.toId())) {
+      getDatasetClient().delete(namespace.dataset(datasetSpecSummary.getName()).toId());
+    }
+    ArtifactClient artifactClient = new ArtifactClient(getClientConfig(), getRestClient());
+    for (ArtifactSummary artifactSummary : artifactClient.list(namespace.toId())) {
+      if (ArtifactScope.USER == artifactSummary.getScope()) {
```

Comment:
doesn't ``ArtifactClient#list`` allow you to specify if system artifacts should be included in the result?